graylog2-web-interface/src/routing/AppFacade.jsx

```diff
@@ -24,38 +22,58 @@ const LoginPage = loadAsync(() => import(/* webpackChunkName: "LoginPage" */ 'pa
 const LoadingPage = loadAsync(() => import(/* webpackChunkName: "LoadingPage" */ 'pages/LoadingPage'));
 const LoggedInPage = loadAsync(() => import(/* webpackChunkName: "LoggedInPage" */ 'pages/LoggedInPage'));
 
-const AppFacade = createReactClass({
-  displayName: 'AppFacade',
-  mixins: [Reflux.connect(SessionStore), Reflux.connect(ServerAvailabilityStore), Reflux.connect(CurrentUserStore)],
-
-  componentDidMount() {
-    this.interval = setInterval(ServerAvailabilityStore.ping, 20000);
-  },
-
-  componentWillUnmount() {
-    if (this.interval) {
-      clearInterval(this.interval);
-    }
-  },
-
-  render() {
-    const { currentUser, server, sessionId } = this.state;
-    let Page = <LoggedInPage />;
-
-    if (!server.up) {
-      Page = <ServerUnavailablePage server={server} />;
-    } else if (!sessionId) {
-      Page = <LoginPage />;
-    } else if (!currentUser) {
-      Page = <LoadingPage text="We are preparing Graylog for you..." />;
-    }
-
-    return (
-      <GraylogThemeProvider>
-        {Page}
-      </GraylogThemeProvider>
-    );
-  },
-});
-
-export default AppFacade;
+const SERVER_PING_TIMEOUT = 20000;
+
+export const AppFacade = ({ currentUser, server, sessionId }) => {
+  let interval;
+  let Page;
+
+  useEffect(() => {
+    interval = setInterval(ServerAvailabilityStore.ping, SERVER_PING_TIMEOUT);
+    return () => {
```

Comment:
We can shorten this to

```suggestion
    return () => clearInterval(interval);
```

a) we know that `interval` has been assigned before, b) even if `setInterval` returns something invalid because of an error, `clearInterval` does not raise an error.